Services/SelectedStopPointManager.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace CanalTP\MttBundle\Services;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Symfony\Component\Serializer\Serializer;
+
+use CanalTP\MttBundle\Entity\LineTimetable;
+use CanalTP\MttBundle\Entity\SelectedStopPoint;
+
+/**
+ * Class SelectedStopPointManager
+ * @package CanalTP\MttBundle\Services
+ */
+class SelectedStopPointManager
+{
+    private $om = null;
+    private $repository = null;
+    private $serializer = null;
+    private $navitia = null;
+
+    /**
+     * Constructor
+     *
+     * @param ObjectManager $om
+     */
+    public function __construct(ObjectManager $om, Navitia $navitia, Serializer $serializer)
+    {
+        $this->om = $om;
+        $this->repository = $this->om->getRepository('CanalTPMttBundle:SelectedStopPoint');
+        $this->serializer = $serializer;
+        $this->navitia = $navitia;
+    }
+
+    /**
+     * Preparing a list with available stop points and their selection state
+     *
+     * @param mixed $navitiaStopPoints
+     * @param Collection $selectedStopPoints
+     */
+    private function tagAvailableStopPoints($navitiaStopPoints, Collection $selectedStopPoints)
+    {
+        $stopPointsIds = $selectedStopPoints->map(
+            function($entity) {
+                return $entity->getExternalStopPointId();
+            }
+        )->toArray();
+
+        $stopPoints = array();
+        foreach ($navitiaStopPoints as $stopPoint)
+        {
+            $stopPoint->selected = in_array($stopPoint->id, $stopPointsIds);
+            $stopPoints[$stopPoint->id] = $stopPoint;
+        }
+
+        return $stopPoints;
+    }
+
+    /**
+     * Preparing available stop points
+     *
+     * @param string $externalCoverageId
+     * @param string $externalRouteId
+     * @param LineTimetable $lineTimetable
+     * @param mixed $routes
+     */
+    public function prepareStopsSelection($externalCoverageId, $externalRouteId, LineTimetable $lineTimetable, $routes)
+    {
+        $selectedStopPoints = $lineTimetable->getSelectedStopPointsByRoute($externalRouteId);
+
+        $navitiaStopPoints = $this->navitia->getRouteData(
+            $externalRouteId,
+            $externalCoverageId
+        )->stop_points;
+
+        // Reversing selection if there are 2 routes and only one with selected stop points
+        $reversed = false;
+        if ($selectedStopPoints->isEmpty() && count($routes) == 2)
+        {
+            $revExternalRouteId = array_pop(array_filter($routes,
```

Comment:
I have the following error 
Runtime Notice: Only variables should be passed by reference

as array_pop expects to have an variable not a function.


```php
$filteredRoutes = array_filter($routes,
    function ($route) use ($externalRouteId) {
        return $route->id != $externalRouteId;
    }
);
$revExternalRouteId = array_pop($filteredRoutes)->id;
```